src/coredump/coredump.c

```diff
@@ -560,6 +560,90 @@ static int compose_open_fds(pid_t pid, char **open_fds) {
         return 0;
 }
 
+static int get_process_ns(pid_t pid, const char *namespace, ino_t *ns) {
+        const char *p;
+        struct stat stbuf;
+        _cleanup_close_ int proc_ns_dir_fd;
+
+        p = procfs_file_alloca(pid, "ns");
+
+        proc_ns_dir_fd = open(p, O_DIRECTORY);
+        if (proc_ns_dir_fd < 0)
+                return -errno;
+
+        if (fstatat(proc_ns_dir_fd, namespace, &stbuf, /* flags */0) != 0)
+                return -errno;
+
+        *ns = stbuf.st_ino;
+        return 0;
+}
+
+/* Find the first process in the parent hierarchy which has different mount
+ * namespace and call it the pid's container parent. */
+static int get_process_container_parent_pid(pid_t pid, pid_t *container_pid) {
+        pid_t cpid = pid, ppid = 0;
+        ino_t proc_mntns;
+        int r = 0;
+
+        r = get_process_ns(pid, "mnt", &proc_mntns);
+        if (r != 0)
```

Comment:
I think < 0 would be better here, too, just because that's usually how we do this, and doing this differently here just raises eyebrows, and the question whether there's a specific reason to have a different check here (i don't think there is...)